lib/verify/response.rb

```diff
@@ -26,34 +26,45 @@ def verified?
     def claim_parameters
       return {} unless verified?
 
-      {
+      identity_paramteters.merge(
+        verified_fields: verified_fields
+      )
+    end
+
+    def scenario
+      @scenario ||= parameters["scenario"]
+    end
+
+    private
+
+    def identity_paramteters
+      @identity_paramteters ||= {
         full_name: full_name,
         address_line_1: address_lines[0],
         address_line_2: address_lines[1],
         address_line_3: address_lines[2],
         postcode: address.fetch("postCode"),
-        date_of_birth: parameters.fetch("attributes").fetch("datesOfBirth").first.fetch("value"),
+        date_of_birth: attributes.dig("datesOfBirth").first&.dig("value"),
+        gender: gender,
       }
     end
 
-    def scenario
-      @scenario ||= parameters["scenario"]
+    def verified_fields
+      identity_paramteters.reject { |k, v| v.blank? }.keys
     end
 
-    private
-
     def address
-      @address ||= most_recent_verified_value(parameters.fetch("attributes").fetch("addresses"))
+      @address ||= most_recent_verified_value(attributes.dig("addresses"))
     end
 
     def address_lines
       @address_lines ||= address.fetch("lines")
     end
 
     def full_name
-      first_name = most_recent_verified_value(parameters.fetch("attributes").fetch("firstNames"))
-      middle_name = most_recent_verified_value(parameters.fetch("attributes").fetch("middleNames"))
-      surname = most_recent_verified_value(parameters.fetch("attributes").fetch("surnames"))
+      first_name = most_recent_verified_value(attributes.dig("firstNames"))
+      middle_name = most_recent_verified_value(attributes.dig("middleNames"))
+      surname = most_recent_verified_value(attributes.dig("surnames"))
```

Comment:
Is it true that we always get more than one name for a user? My name, for instance, is a single word, though Verify may not support this.

Reply:
As far as the Verify documentation goes, we should always get back a surnames value. This might be a question we can only answer by actually hitting Verify for real. I'd suggest we'll be in a better position to tackle this as a separate iteration once we've got the basics covered and can see how Verify behaves in this case.

Reply:
I've added a card to the backlog to look at unusual data in Verify.